File: April 2024 GFG SOLUTION/April-22.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
  public:
    int minRow(int n, int m, vector<vector<int>> a) {
        int res = 1;
        int min = INT_MAX;
        for(int i = 0; i < n; i++) {
            int count = 0;
            for(int j = 0; j < m; j++) {
                if(a[i][j] == 1) {
                    count++;
                }
            }
            
            int minn = std::min(count, min);
            if(minn != min) {
                res = i + 1;
                min = minn;
            }
        }
        return res;
    }
};


//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N, M;
        cin >> N >> M;
        vector<vector<int>> A(N, vector<int>(M));
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < M; j++) {
                cin >> A[i][j];
            }
        }
        Solution ob;
        cout << ob.minRow(N, M, A) << "\n";
    }
}
// } Driver Code Ends

File: April 2024 GFG SOLUTION/April-22.py
#User function Template for python3

class Solution:
    def minRow(self, n, m, a):
        res = 1
        min_val = math.inf
        for i in range(n):
            count = 0
            for j in range(m):
                if a[i][j] == 1:
                    count += 1
            minn = min(count, min_val)
            if minn != min_val:
                res = i + 1
                min_val = minn
        return res

 # Driver Code Starts
#Initial Template for Python 3

import math
        
if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        N,M=map(int,input().strip().split(" "))
        A=[]
        for i in range(N):
            B=list(map(int,input().strip().split(" ")))
            A.append(B)
        ob=Solution()
        print(ob.minRow(N,M,A))
# } Driver Code Ends
